src/test/java/io/jenkins/infra/repository_permissions_updater/JiraAPITest.java

```diff
@@ -0,0 +1,227 @@
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.LoggerContext;
+import io.jenkins.infra.repository_permissions_updater.helper.HttpUrlStreamHandler;
+import io.jenkins.infra.repository_permissions_updater.helper.MemoryAppender;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.net.HttpURLConnection;
+import java.net.ProtocolException;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLStreamHandlerFactory;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Properties;
+import java.util.function.Supplier;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+class JiraAPITest {
+
+    private static MemoryAppender memoryAppender;
+    private static HttpUrlStreamHandler httpUrlStreamHandler;
+
+    private static final Supplier<String> JIRA_URL = () -> System.getProperty("jiraUrl", "https://issues.jenkins.io");
+    private static final  Supplier<String> JIRA_USE_URL = () -> JIRA_URL.get() + "/rest/api/2/user";
+    private static final  Supplier<String> JIRA_USER_QUERY = () -> JIRA_USE_URL.get() + "?username=%s";
+    private static final Supplier<String> JIRA_COMPONENTS_URL = () -> JIRA_URL.get() + "/rest/api/2/project/JENKINS/components";
+    private static URLStreamHandlerFactory urlStreamHandlerFactory;
+    private Properties backup;
+
+
+    @BeforeAll
+    public static void setup() {
+        Logger logger = (Logger) LoggerFactory.getLogger(JiraImpl.class);
+        memoryAppender = new MemoryAppender();
+        memoryAppender.setContext((LoggerContext) LoggerFactory.getILoggerFactory());
+        logger.setLevel(Level.DEBUG);
+        logger.addAppender(memoryAppender);
+        memoryAppender.start();
+
+        urlStreamHandlerFactory = mock(URLStreamHandlerFactory.class);
+        httpUrlStreamHandler = new HttpUrlStreamHandler();
+        when(urlStreamHandlerFactory.createURLStreamHandler("http")).thenReturn(httpUrlStreamHandler);
+        when(urlStreamHandlerFactory.createURLStreamHandler("https")).thenReturn(httpUrlStreamHandler);
+        URL.setURLStreamHandlerFactory(urlStreamHandlerFactory);
+    }
+
+    @BeforeEach
+    public void reset() throws NoSuchFieldException, IllegalAccessException {
+        backup = new Properties();
+        backup.putAll(System.getProperties());
+
+        Field instance = JiraAPI.class.getDeclaredField("INSTANCE");
+        instance.setAccessible(true);
+        instance.set(null, null);
```

Comment:
⛔ do not use reflection in tests for code that you control, either expose the code by marking it as public or make it package-private and let the test access it